### PR TITLE
More updates to CHANGES and NEWS for the 1.1.1 release

### PR DESCRIPTION
**Pull request by mattcaswell**

### CODE REVIEW

File: NEWS [hunk 1]
```diff
@@ -21,7 +33,7 @@
       o Support for various new cryptographic algorithms including:
           o SHA3
           o SHA512/224 and SHA512/256
-          o EdDSA (including Ed25519 and Ed448)
+          o EdDSA (both Ed25519 and Ed448) and including X509 and TLS support
```
Comment (kaduk) [inline NEWS]: This "and" reads oddly.

File: NEWS [hunk 2]
```diff
@@ -30,6 +42,10 @@
           o SipHash
           o ARIA (including TLS support)
       o Significant Side-Channel attack security improvements
+      o Add a new ClientHello callback. Provides a callback interface that gives
+        the application the ability to adjust the nascent SSL object at the
+        earliest stage of ClientHello processing, immediately after extensions
+        have been collected but before they have been processed.
```
Comment (kaduk) [inline NEWS]: Gosh, this text even sounds like it was written by me ;)

One potentially noteworthy aspect (and a recurring feature request, IIUC) is that this allows the callback to affect TLS version negotiation, something that was previously impossible for application code to do.  So maybe, "In particular, this callback can adjust the supported TLS versions in response to the contents of the ClientHello".  But maybe that's too much detail; I think I'm too close to the work to be an objective arbiter.

File: NEWS [hunk 3]
```diff
@@ -7,7 +7,19 @@
 
   Major changes between OpenSSL 1.1.0i and OpenSSL 1.1.1 [in pre-release]
 
-      o Support for TLSv1.3 added
+      o Support for TLSv1.3 added (see https://wiki.openssl.org/index.php/TLS1.3
+        for further important information). The TLSv1.3 implementation includes:
+          o Full support for RFC8446 (TLSv1.3) on by default
```
Comment (kaduk) [inline NEWS]: I'm a little leery of the word "full", as it's a large document and I don't think that we explicitly verified compliance with all MUSTs, SHOULDs, MAYs, etc. (and I could see someone reading "full" as including all the optional stuff, too).
But I also can't think of anything off the top of my head that we didn't implement, so maybe it's fine to leave it as-is.

File: NEWS (hunk 2 above)
Comment (levitte) [inline NEWS]: Keep it a one-liner, please.  If we start copying straight from CHANGES, it becomes ridiculous.  NEWS is meant to be short (one-line or close to) summaries

File: NEWS (hunk 3 above)
Comment (mattcaswell) [reply NEWS]: We haven't done all optionals (no FFDHE springs to mind). But we should be fully compliant. Any missing MUSTs would be a bug. So perhaps reword to "Fully compliant implementation of RFC8446 (TLSv1.3) on by default"

File: NEWS (hunk 2 above)
Comment (mattcaswell) [reply NEWS]: I'll add that to CHANGES and slim down the NEWS entry as per the comment by @levitte.

**Comment by mattcaswell** [comment]
Fixup commit pushed addressing the feedback so far.

**Comment by mattcaswell** [comment]
Pushed.

**Pull request closed without merging**